spring-boot-dixit/src/test/java/tw/wally/dixit/DixitControllerTest.java

```diff
@@ -0,0 +1,277 @@
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static tw.wally.dixit.utils.StreamUtils.*;
+
+@ContextConfiguration(classes = {DixitApplication.class, DixitControllerTest.DixitConfiguration.class})
+public class DixitControllerTest extends AbstractSpringBootTest {
+    protected static final String API_PREFIX = "/api/dixit";
+    private static final String ROOM_ID = "roomId";
+    private static final String DIXIT_ID = "dixitId";
+    private static final String DIXIT_GAMER = "dixitGamer";
+    private static final String PHRASE = "phrase";
+
+    @Autowired
+    private DixitRepository dixitRepository;
+
+    @Configuration
+    public static class DixitConfiguration {
+
+        @Bean
+        @Primary
+        public TokenService testTokenService() {
+            return new FakeTokenService();
+        }
+
+        @Bean
+        @Primary
+        public CardRepository testCardRepository() {
+            return new FakeCardRepository();
+        }
+    }
+
+    @Test
+    public void WhenCreateDixitWithFourGamers_ThenShouldSuccess() throws Exception {
+        createDixitWithGamers(4)
+                .andExpect(status().isOk());
+
+        assertTrue(dixitRepository.findDixitById(DIXIT_ID).isPresent());
+    }
+
+    @Test
+    public void WhenCreateDixitWithEightGamers_ThenShouldFail() throws Exception {
+        createDixitWithGamers(8)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenDixitStarted_WhenStorytellerTellStory_ThenCurrentRoundShouldHaveStory() throws Exception {
+        Dixit dixit = createDixitWithGamersAndGet(6);
+        assertNull(dixit.getCurrentRound().getStory());
+
+        var storyteller = dixit.getCurrentStoryteller();
+        tellStory(dixit.getNumberOfRounds(), storyteller)
+                .andExpect(status().isOk());
+
+        assertStorytellerTellStory(storyteller);
+    }
+
+    @Test
+    public void GivenDixitStarted_WhenGuesserTellStory_ThenShouldFail() throws Exception {
+        Dixit dixit = createDixitWithGamersAndGet(5);
+        assertNull(dixit.getCurrentRound().getStory());
+
+        Player guesser = dixit.getCurrentGuessers().get(0);
+        tellStory(dixit.getNumberOfRounds(), guesser)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void GivenDixitStartedAndFirstRoundStarted_WhenStorytellerTellStoryAtSecondRound_ThenShouldFail() throws Exception {
+        Dixit dixit = createDixitWithGamersAndGet(6);
+        assertEquals(1, dixit.getNumberOfRounds());
+
+        Player storyteller = dixit.getCurrentStoryteller();
+        tellStory(2, storyteller)
+                .andExpect(status().isBadRequest());
+    }
```

Comment:
Is it better to extract `1` to a constant `FIRST_ROUND` and `2` to `SECOUND_RONUD`? 
This is just a suggestion.